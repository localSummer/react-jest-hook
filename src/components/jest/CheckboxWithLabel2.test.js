import React from 'react';
import {shallow, configure} from 'enzyme';
import CheckboxWithLabel from './CheckboxWithLabel';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('CheckboxWithLabel changes the text after click', () => {
  const checkbox = shallow(<CheckboxWithLabel labelOn="On" labelOff="Off" />);
  expect(checkbox.text()).toEqual('Off');
  expect(checkbox.text()).toMatchSnapshot();

  checkbox.find('input').simulate('change');

  expect(checkbox.text()).toEqual('On');
  expect(checkbox.text()).toMatchSnapshot();
})


