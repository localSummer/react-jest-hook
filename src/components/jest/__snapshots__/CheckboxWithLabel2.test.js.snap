// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel changes the text after click 1`] = `"Off"`;

exports[`CheckboxWithLabel changes the text after click 2`] = `"On"`;
